File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/ui/guides/CustomizingScrollBehavior.java
package io.getstream.chat.docs.java.ui.guides;

import io.getstream.chat.android.ui.helper.TransformStyle;

/**
 * [Customizing MessageListView Scrolling Behavior](https://getstream.io/chat/docs/sdk/android/ui/guides/customizing-message-list-scroll-behavior/)
 */
public class CustomizingScrollBehavior {

    public void customization() {
        TransformStyle.setMessageListStyleTransformer(defaultViewStyle -> {
            // Customize the style
            return defaultViewStyle;
        });
    }
}

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/ui/guides/CustomizingScrollBehavior.kt
package io.getstream.chat.docs.kotlin.ui.guides

import io.getstream.chat.android.ui.helper.StyleTransformer
import io.getstream.chat.android.ui.helper.TransformStyle

/**
 * [Customizing MessageListView Scrolling Behavior](https://getstream.io/chat/docs/sdk/android/ui/guides/customizing-message-list-scroll-behavior/)
 */
class CustomizingScrollBehavior {

    fun customization() {
        TransformStyle.messageListStyleTransformer = StyleTransformer { defaultViewStyle ->
            defaultViewStyle.copy(
                disableScrollWhenShowingDialog = false
            )
        }
    }
}
